tests/test_scheduler_local.py

```
@@ -0,0 +1,369 @@
+    def test_naive_treated_as_utc(self):
+        dt = parse_iso_to_utc("2026-04-28T15:00:00")
+        assert dt.tzinfo == UTC
+        assert dt.hour == 15
+
+    def test_offset_normalized(self):
+        dt = parse_iso_to_utc("2026-04-28T15:00:00-05:00")
+        assert dt.tzinfo == UTC
+        assert dt.hour == 20  # 15 EST → 20 UTC
+
+    def test_malformed_raises(self):
+        with pytest.raises(ValueError, match="Invalid isoformat|could not convert"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)